code/modules/photography/photos/frame.dm

```diff
@@ -113,6 +105,15 @@
 		framed = P
 		update_appearance()
 
+/// Given a photo (or null), will change the contained picture, and queue a persistent save.
+/obj/structure/sign/picture_frame/proc/set_and_save_framed(obj/item/photo/photo)
+	framed = photo
+
+	if (isnull(persistence_id))
+		return
+
+	SSpersistence.photo_frames_database.set_key(persistence_id, photo.picture?.id)
```

Comment:
`photo` can be null but there is no null check for `photo.picture`

Reply:
```suggestion
/// Given a photo (or null), will change the contained picture, and queue a persistent save.
/obj/structure/sign/picture_frame/proc/set_and_save_framed(obj/item/photo/photo)
	framed = photo

	if (isnull(persistence_id))
		return

	SSpersistence.photo_frames_database.set_key(persistence_id, photo?.picture?.id)
```